p4-16/psa/PSA.mdk

```diff
@@ -970,80 +959,84 @@ The partial program below demonstrates how to clone a packet.
- it effectively increase the amount of operations on the packet by N
- folds, where N is the number of times the packet is resubmitted.
- Another use case is deploy multiple packet processing algorithms on
- the same packet. For example, the original packet can be parsed,
- modified and resubmitted in the first pass. Then, the resubmitted
- packet can be parsed and modified using a different algorithm.
+Packet resubmission happens at the end of the ingress pipeline. When a
+packet is resubmitted, the packet finishes the ingress pipeline
+processing and re-enters the ingress parser without being deparsed. In
+other words, the resubmitted packet has the same header and payload as
+the original packet. The `ingress_port` of the resubmitted packet is
+the same as the original packet. The `packet_path` of the resubmitted
+packet is changed to `RESUBMIT`.
+
+The ingress parser distinguishes the resubmitted packet from the
+original packet with the `packet_path` field in
+`ingress_parser_intrinsic_metadata_t`. The ingress parser can choose a
+different algorithm to parse the resubmitted packet. Similarly, the
+ingress pipeline can choose to process the resubmitted packet with
+different actions as opposed to the ones used to process the original
+packet. Further, if a target permits the same packet to be resubmitted
+multiple times, the user program can distinguish the packet
+resubmitted the first time, or second time, by the extra metadata
+associated with the packet. Note the maximum number of packet
+resubmission for a single packet is target-dependent, see section
+[#packet-paths].
+
+PSA specifies that the resubmit operation can only be used in the
+ingress pipeline. The egress pipeline cannot resubmit packets. As
+described in Section [#packet-paths], there is no mandated mechanism
+in PSA to prevent a single received packet from creating packets that
+continue to recirculate, resubmit, or clone from egress to egress
+indefinitely. However, targets may impose limits on the number of
+resubmissions, recirculations, or clones.
+
+One use case of packet resubmission is to increase the capacity and
+flexibility of the packet processing pipeline. For example, because
+the same packet is processed by the ingress pipeline multiple times,
+it effectively increase the amount of operations on the packet by N
+folds, where N is the number of times the packet is resubmitted.
+
+Another use case is to deploy multiple packet processing algorithms on
+the same packet. For example, the original packet can be parsed and
+resubmitted in the first pass with additional metadata to select one
+of the algorithms. Then, the resubmitted packet can be parsed,
+modified and deparsed using the selected algorithm.
 
 To facilitate communication between the first pass and the second pass
-through the ingress pipeline, the resubmission mechanism supports
+through the ingress pipeline, the resubmission mechanism supportsr
```

Comment:
typo?

Reply:
Yes.  I did the conflict resolution using Github's web interface, not normal git commands.  I'm happy to fix this in a later PR, unless you find it easy to figure out how to use git to pull in those changes, and add another commit on top.